src/com/android/calendar/agenda/AgendaByDayAdapter.java

```diff
@@ -235,12 +235,24 @@ public View getView(int position, View convertView, ViewGroup parent) {
             holder.dateView.setText(dateViewText);
 
             // Set the background of the view, it is grayed for day that are in the past and today
-            if (row.mDay > mTodayJulianDay) {
-                agendaDayView.setBackgroundResource(R.drawable.agenda_item_bg_primary);
-                holder.grayed = false;
+
+            if (theme.equals("dark")) {
+                if (row.mDay > mTodayJulianDay) {
+                    agendaDayView.setBackgroundResource(R.drawable.agenda_item_bg_primary_dark);
```

Comment:
Change to:
`agendaDayView.setBackgroundResource(theme.equals("dark") ? R.drawable.agenda_item_bg_primary_dark : R.drawable.agenda_item_bg_primary);`